### ISSUE TITLE
ImportError: cannot import name 'CompilationException' from 'dbt.exceptions'

### ISSUE THREAD

**Issue by seve-martinez** [sqlfluff/sqlfluff#4325]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I tried to lint a simple file using the VS Code extension. I have a custom .sqlfluff (posted below) that was successfully found by sqlfluff.

sqlfluff crashed, complaining of an ImportError from dbt.

```
Traceback (most recent call last):
  File "/Users/smartinez/.pyenv/versions/3.7.10/bin/sqlfluff", line 5, in <module>
    from sqlfluff.cli.commands import cli
  File "/Users/smartinez/.pyenv/versions/3.7.10/lib/python3.7/site-packages/sqlfluff/cli/commands.py", line 533, in <module>
    ) -> None:
  File "/Users/smartinez/.pyenv/versions/3.7.10/lib/python3.7/site-packages/sqlfluff/cli/commands.py", line 218, in core_options
    get_plugin_manager().hook.get_templaters()
  File "/Users/smartinez/.pyenv/versions/3.7.10/lib/python3.7/site-packages/sqlfluff/core/plugin/host.py", line 12, in get_plugin_manager
    pm.load_setuptools_entrypoints(project_name)
  File "/Users/smartinez/.pyenv/versions/3.7.10/lib/python3.7/site-packages/pluggy/_manager.py", line 287, in load_setuptools_entrypoints
    plugin = ep.load()
  File "/Users/smartinez/.pyenv/versions/3.7.10/lib/python3.7/site-packages/importlib_metadata/__init__.py", line 208, in load
    module = import_module(match.group('module'))
  File "/Users/smartinez/.pyenv/versions/3.7.10/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/Users/smartinez/.pyenv/versions/3.7.10/lib/python3.7/site-packages/sqlfluff_templater_dbt/__init__.py", line 3, in <module>
    from sqlfluff_templater_dbt.templater import DbtTemplater
  File "/Users/smartinez/.pyenv/versions/3.7.10/lib/python3.7/site-packages/sqlfluff_templater_dbt/templater.py", line 17, in <module>
    from dbt.exceptions import (
ImportError: cannot import name 'CompilationException' from 'dbt.exceptions' (/Users/smartinez/.pyenv/versions/3.7.10/lib/python3.7/site-packages/dbt/exceptions.py)
```

This appears to be a direct impact of [this PR](https://sourcegraph.com/github.com/dbt-labs/dbt-core/-/commit/eb200b46877c31469f410b7d47dfd245473ed7cb) from dbt-core which renamed all `Exception` to `Error`.

This includes the `CompilationException` that SqlFluff is looking for.

Here is the current dbt implementation: https://sourcegraph.com/github.com/dbt-labs/dbt-core@82d9b2fa87e4bc4953b82be9baa2a3d9e951b83e/-/blob/core/dbt/exceptions.py?L192

### Expected Behaviour

The file would lint successfully.

### Observed Behaviour

This appears to be a direct impact of [this PR](https://sourcegraph.com/github.com/dbt-labs/dbt-core/-/commit/eb200b46877c31469f410b7d47dfd245473ed7cb) from dbt-core which renamed all `Exception` to `Error`.

This includes the `CompilationException` that SqlFluff is looking for.

Here is the current dbt implementation: https://sourcegraph.com/github.com/dbt-labs/dbt-core@82d9b2fa87e4bc4953b82be9baa2a3d9e951b83e/-/blob/core/dbt/exceptions.py?L192

### How to reproduce

Create a model:

```
select created_at::date, count(user_id) as new_users_by_date
from {{ ref("cbdbt__dim_users") }}
where created_at > {{ dbt_date.n_days_ago(7) }}
group by 1

```

Run a lint:

`sqlfluff fix --force models/cbdbt/cbdbt__fct_new_users/cbdbt__fct_new_users.sql`

### Dialect

Snowflake

### Version

```
sqlfluff==1.4.5
sqlfluff-templater-dbt==1.4.5
```

### Configuration
```
[sqlfluff]
# verbose is an integer (0-2) indicating the level of log output
verbose = 0
# Turn off color formatting of output
nocolor = False
# Supported dialects https://docs.sqlfluff.com/en/stable/dialects.html
# Or run 'sqlfluff dialects'
dialect = snowflake
# One of [raw|jinja|python|placeholder]
templater = dbt
# Comma separated list of rules to check, default to all
rules = all
# Comma separated list of rules to exclude, or None
exclude_rules = L031,L057,L051,L060
# The depth to recursively parse to (0 for unlimited)
recurse = 0
# Below controls SQLFluff output, see max_line_length for SQL output
output_line_length = 80
# Number of passes to run before admitting defeat
runaway_limit = 100
# Ignore errors by category (one or more of the following, separated by commas: lexing,linting,parsing,templating)
ignore = None
# Ignore linting errors found within sections of code coming directly from
# templated code (e.g. from within Jinja curly braces. Note that it does not
# ignore errors from literal code found within template loops.
ignore_templated_areas = True
# can either be autodetect or a valid encoding e.g. utf-8, utf-8-sig
encoding = autodetect
# Ignore inline overrides (e.g. to test if still required)
disable_noqa = False
# Comma separated list of file extensions to lint
# NB: This config will only apply in the root folder
sql_file_exts = .sql,.sql.j2,.dml,.ddl
# Allow fix to run on files, even if they contain parsing errors
# Note altering this is NOT RECOMMENDED as can corrupt SQL
fix_even_unparsable = False
# Very large files can make the parser effectively hang.
# This limit skips files over a certain character length
# and warns the user what has happened.
# Set this to 0 to disable.
large_file_skip_char_limit = 20000

[sqlfluff:indentation]
# See https://docs.sqlfluff.com/en/stable/indentation.html
indented_joins = False
indented_ctes = False
indented_using_on = True
indented_on_contents = True
template_blocks_indent = True

[sqlfluff:templater]
unwrap_wrapped_queries = True

[sqlfluff:templater:jinja]
apply_dbt_builtins = True

[sqlfluff:templater:dbt]
project_dir = .
profiles_dir = .
profile = snowflake-dbt
target = prod

[sqlfluff:templater:jinja:macros]
# Macros provided as builtins for dbt projects
dbt_ref = {% macro ref(model_ref) %}{{model_ref}}{% endmacro %}
dbt_source = {% macro source(source_name, table) %}{{source_name}}_{{table}}{% endmacro %}
dbt_config = {% macro config() %}{% for k in kwargs %}{% endfor %}{% endmacro %}
dbt_var = {% macro var(variable, default='') %}item{% endmacro %}
dbt_is_incremental = {% macro is_incremental() %}True{% endmacro %}


[sqlfluff:templater:dbt:context]
# Put vars defined in dbt_project.yml here


[sqlfluff:templater:jinja:context]
# Put standard jinja params here

# Some rules can be configured directly from the config common to other rules
[sqlfluff:rules]
tab_space_size = 2
max_line_length = 80
indent_unit = space
# comma_style = leading
allow_scalar = True
single_table_references = consistent
unquoted_identifiers_policy = all

# Some rules have their own specific config
[sqlfluff:rules:L003]
hanging_indents = True


[sqlfluff:rules:L010]
# Keywords
capitalisation_policy = upper
# Comma separated list of words to ignore for this rule
ignore_words = None
ignore_words_regex = None

[sqlfluff:rules:L011]
# Aliasing preference for tables
aliasing = explicit

[sqlfluff:rules:L012]
# Aliasing preference for columns
aliasing = explicit

[sqlfluff:rules:L014]
# Unquoted identifiers
extended_capitalisation_policy = lower
# Comma separated list of words to ignore for this rule
ignore_words = None
ignore_words_regex = None

[sqlfluff:rules:L016]
# Line length
ignore_comment_lines = False
ignore_comment_clauses = False

[sqlfluff:rules:L027]
# Comma separated list of words to ignore for this rule
ignore_words = None
ignore_words_regex = None

[sqlfluff:rules:L026]
# References must be in FROM clause
# Disabled for some dialects (e.g. bigquery)
force_enable = False

[sqlfluff:rules:L028]
# References must be consistently used
# Disabled for some dialects (e.g. bigquery)
force_enable = False

[sqlfluff:rules:L029]
# Keywords should not be used as identifiers.
unquoted_identifiers_policy = aliases
quoted_identifiers_policy = none
# Comma separated list of words to ignore for this rule
ignore_words = None
ignore_words_regex = None

[sqlfluff:rules:L030]
# Function names
extended_capitalisation_policy = upper
# Comma separated list of words to ignore for this rule
ignore_words = None
ignore_words_regex = None

[sqlfluff:rules:L031]
# Avoid table aliases in from clauses and join conditions.
# Disabled for some dialects (e.g. bigquery)
force_enable = False

[sqlfluff:rules:L036]
wildcard_policy = single

[sqlfluff:rules:L038]
# Trailing commas
select_clause_trailing_comma = forbid

[sqlfluff:rules:L040]
# Null & Boolean Literals
capitalisation_policy = upper
# Comma separated list of words to ignore for this rule
ignore_words = None
ignore_words_regex = None

[sqlfluff:rules:L042]
# By default, allow subqueries in from clauses, but not join clauses
forbid_subquery_in = join

[sqlfluff:rules:L047]
# Consistent syntax to count all rows
prefer_count_1 = False
prefer_count_0 = False

[sqlfluff:rules:L051]
# Fully qualify JOIN clause
fully_qualify_join_types = inner

[sqlfluff:rules:L052]
# Semi-colon formatting approach
multiline_newline = False
require_final_semicolon = False

[sqlfluff:rules:L054]
# GROUP BY/ORDER BY column references
group_by_and_order_by_style = consistent

[sqlfluff:rules:L057]
# Special characters in identifiers
unquoted_identifiers_policy = all
quoted_identifiers_policy = all
allow_space_in_identifier = False
additional_allowed_characters = ""
ignore_words = None
ignore_words_regex = None

[sqlfluff:rules:L059]
# Policy on quoted and unquoted identifiers
prefer_quoted_identifiers = False
ignore_words = None
ignore_words_regex = None
force_enable = False

[sqlfluff:rules:L062]
# Comma separated list of blocked words that should not be used
blocked_words = None
blocked_regex = None

[sqlfluff:rules:L063]
# Data Types
extended_capitalisation_policy = consistent
# Comma separated list of words to ignore for this rule
ignore_words = None
ignore_words_regex = None

[sqlfluff:rules:L064]
# Consistent usage of preferred quotes for quoted literals
preferred_quoted_literal_style = consistent
# Disabled for dialects that do not support single and double quotes for quoted literals (e.g. Postgres)
force_enable = False

[sqlfluff:rules:L066]
min_alias_length = None
max_alias_length = None
```
### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by barrywhart** [comment]
This is a duplicate issue. It's been fixed in 2.0.0a4. Or you can use an older version of dbt-core.

Or create an issue against dbt-core, for renaming all their exceptions in a _minor release_ and breaking things.

**Comment by elyobo** [comment]
Oof, that's bad form from dbt.
